From cd749ac80f5d41f74c9de5a60f01ac4b065566dd Mon Sep 17 00:00:00 2001
From: Hanlu Li <heuleehanlu@gmail.com>
Date: Fri, 18 Apr 2025 10:18:20 +0800
Subject: [PATCH 1050/1058] LAT, fix: Fix compilation error caused by duplicate
 macro definition
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This fixes the following compilation failure on AOSC OS:

../linux-user/syscall_defs.h:1533:9: 错误：“MAP_HUGE_2MB”重定义 [-Werror]
 1533 | #define MAP_HUGE_2MB     (21 << HUGETLB_FLAG_ENCODE_SHIFT)
      |         ^~~~~~~~~~~~
In file included from /usr/include/bits/mman.h:32,
                 from /usr/include/sys/mman.h:41,
                 from /home/lauren/lat/include/sysemu/os-posix.h:29,
                 from /home/lauren/lat/include/qemu/osdep.h:161,
                 from ../linux-user/i386/signal.c:19:
/usr/include/bits/mman-linux.h:70:9: 附注：这是先前定义的位置
   70 | #define MAP_HUGE_2MB    (21 << MAP_HUGE_SHIFT)
      |         ^~~~~~~~~~~~
../linux-user/syscall_defs.h:1534:9: 错误：“MAP_HUGE_1GB”重定义 [-Werror]
 1534 | #define MAP_HUGE_1GB     (30 << HUGETLB_FLAG_ENCODE_SHIFT)
      |         ^~~~~~~~~~~~
/usr/include/bits/mman-linux.h:76:9: 附注：这是先前定义的位置
   76 | #define MAP_HUGE_1GB    (30 << MAP_HUGE_SHIFT)

Signed-off-by: Hanlu Li <heuleehanlu@gmail.com>
---
 linux-user/syscall_defs.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/linux-user/syscall_defs.h b/linux-user/syscall_defs.h
index 6fe3080d38..4e42eec516 100644
--- a/linux-user/syscall_defs.h
+++ b/linux-user/syscall_defs.h
@@ -1530,8 +1530,12 @@ struct target_winsize {
 #include "termbits.h"
 
 #define HUGETLB_FLAG_ENCODE_SHIFT   26
+#ifndef MAP_HUGE_2MB
 #define MAP_HUGE_2MB     (21 << HUGETLB_FLAG_ENCODE_SHIFT)
+#endif
+#ifndef MAP_HUGE_1GB
 #define MAP_HUGE_1GB     (30 << HUGETLB_FLAG_ENCODE_SHIFT)
+#endif
 #define LEGACY_MAP_MASK (TARGET_MAP_SHARED \
         | TARGET_MAP_PRIVATE \
         | TARGET_MAP_FIXED \
-- 
2.49.0

